doc/userguide/rules/intro.rst

```diff
@@ -349,3 +349,201 @@ reassembled streams, TLS-, SSL-, SSH-, FTP- and dcerpc-buffers.
 
 Note that there are some exceptions, e.g. the ``http_raw_uri`` keyword.
 See :ref:`rules-http-uri-normalization` for more information.
+
+
+Rule's Types and Categorization
+-------------------------------
+
+Once parsed, Suricata rules are categorized for performance and further
+processing (as different rule types will be handled by specific engine modules).
+The signature types are defined in `src/detect.h
+<https://github.com/OISF/suricata/blob/master/src/detect.h#L62-L75>`_.
```

Comment:
should we embed this code/enum using the literal block or something?

Reply:
Okie. Thought about leaving it as a link just so people would only check it if they wanted to go more in-depth.

Reply:
my issue with a link like that is that its very fragile. Before we know it, line numbers are off.